{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from urllib.request import Request, urlopen, urlretrieve\n",
    "from IPython.display import display, HTML\n",
    "from bs4 import BeautifulSoup\n",
    "import pandas as pd\n",
    "import requests\n",
    "\n",
    "class HTMLTableParser:\n",
    "    def __init__(self, url):\n",
    "        self.url = url\n",
    "\n",
    "    def find_table(self):\n",
    "        response = requests.get(self.url)\n",
    "        soup = BeautifulSoup(response.text,  \"html.parser\")\n",
    "        table = soup.find_all('table')[0]\n",
    "        return table\n",
    "            \n",
    "\n",
    "    def parse_html_table(self, table):\n",
    "        n_columns = 0\n",
    "        n_rows=0\n",
    "        column_names = []\n",
    "\n",
    "        # Find number of rows and columns\n",
    "        # we also find the column titles if we can\n",
    "        for row in table.find_all('tr'):\n",
    "\n",
    "            # Determine the number of rows in the table\n",
    "            td_tags = row.find_all('td')\n",
    "            if len(td_tags) > 0:\n",
    "                n_rows+=1\n",
    "                if n_columns == 0:\n",
    "                    # Set the number of columns for our table\n",
    "                    n_columns = len(td_tags)\n",
    "\n",
    "            # Handle column names if we find them\n",
    "            th_tags = row.find_all('th')\n",
    "            if len(th_tags) > 0 and len(column_names) == 0:\n",
    "                for th in th_tags:\n",
    "                    column_names.append(th.get_text())\n",
    "\n",
    "        # Safeguard on Column Titles\n",
    "        if len(column_names) > 0 and len(column_names) != n_columns:\n",
    "            raise Exception(\"Column titles do not match the number of columns\")\n",
    "\n",
    "        columns = column_names if len(column_names) > 0 else range(0,n_columns)\n",
    "        df = pd.DataFrame(columns = columns,\n",
    "                          index= range(0, n_rows))\n",
    "        row_marker = 0\n",
    "        for row in table.find_all('tr'):\n",
    "            column_marker = 0\n",
    "            columns = row.find_all('td')\n",
    "            for column in columns:\n",
    "                df.iat[row_marker,column_marker] = column.get_text()\n",
    "                column_marker += 1\n",
    "            if len(columns) > 0:\n",
    "                row_marker += 1\n",
    "        \n",
    "        return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def format_header(x):\n",
    "    if x == 0 :\n",
    "        return 'Date'\n",
    "    if x == 1 :\n",
    "        return 'Etudiant_Directeur'\n",
    "    if x == 2 :\n",
    "        return 'Titre_soutenance'\n",
    "    if x == 3 :\n",
    "        return 'Programme'\n",
    "    if x == 4 :\n",
    "        return 'Grade_obtenu'\n",
    "    \n",
    "def custom_column(table):\n",
    "    header = table.iloc[0]\n",
    "    for i in header.index :\n",
    "        header[i] = format_header(i)\n",
    "    table = table.rename(columns = header)\n",
    "    return table\n",
    "        \n",
    "def get_table(url):\n",
    "    udm_site = HTMLTableParser(url)\n",
    "    table = udm_site.find_table()\n",
    "    tmp = udm_site.parse_html_table(table)\n",
    "    format_tmp = custom_column(tmp)\n",
    "    return format_tmp\n",
    "\n",
    "def format_content(x):\n",
    "    if '\\n' in x :\n",
    "        x = x.replace('\\n', ' / ') \n",
    "        return x\n",
    "    return x"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1. Collect all links from 2000 to now"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "## To update each year\n",
    "start = 2000\n",
    "end = 2020 ## Change for the current year -1\n",
    "\n",
    "phd_links = []\n",
    "master_links = []\n",
    "url = 'https://biochimie.umontreal.ca/etudes/memoires-et-soutenances/'\n",
    "for Y in range(start, end+1):\n",
    "    if 2000 == Y or 2001 == Y :\n",
    "        phd_links.append('{}soutenances-{}/'.format(url, Y))\n",
    "    else:\n",
    "        phd_links.append('{}soutenances-de-these-{}/'.format(url, Y))\n",
    "        master_links.append('{}memoires-de-maitrise-{}/'.format(url, Y))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2. Create table of UdeM bioinformatics PhD students"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in range(len(phd_links)):\n",
    "    if i == 0 :\n",
    "        all_phd_student = get_table(phd_links[i])\n",
    "    else:\n",
    "        tmp = get_table(phd_links[i])\n",
    "        all_phd_student = all_phd_student.append(tmp, sort=True)\n",
    "## All student table\n",
    "all_phd_student = all_phd_student.dropna().reset_index().drop(columns=['index'])\n",
    "all_phd_student['Etudiant_Directeur'] = all_phd_student['Etudiant_Directeur'].apply(format_content)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "27\n"
     ]
    }
   ],
   "source": [
    "bioinfo_phd = all_phd_student[all_phd_student['Programme'] == 'Bio-informatique']\n",
    "print(len(bioinfo_phd))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "                         Date  \\\n",
       "61                 17/12/2007   \n",
       "62                 14/12/2007   \n",
       "65                 07/08/2007   \n",
       "86                 27/08/2009   \n",
       "104                28/09/2010   \n",
       "107                23/06/2010   \n",
       "112                20/01/2010   \n",
       "113                29/01/2010   \n",
       "117                13/12/2011   \n",
       "128                25/07/2011   \n",
       "151                16/04/2012   \n",
       "159                23/07/2013   \n",
       "161                26/06/2013   \n",
       "162                21/06/2013   \n",
       "165                30/05/2013   \n",
       "177                04/07/2014   \n",
       "178                25/06/2014   \n",
       "185                13/02/2014   \n",
       "186                22/01/2014   \n",
       "202                2016-11-29   \n",
       "212                2017-11-20   \n",
       "216                2017-08-22   \n",
       "227                2018-12-07   \n",
       "229                2018-11-07   \n",
       "230                2018-11-06   \n",
       "234                2018-09-20   \n",
       "246  Mardi 1er septembre 2020   \n",
       "\n",
       "                                    Etudiant_Directeur Grade_obtenu  \\\n",
       "61                  Sivakumar Kannan / Gertraud Burger   07/02/2008   \n",
       "62                   Nicolas Rodrigue / Hervé Philippe   07/02/2008   \n",
       "65                  Emmanuelle Permal / François Major   04/10/2007   \n",
       "86                           Yan Zhou / Hervé Philippe   08/10/2009   \n",
       "104            Claudia Laura Kleinman / Hervé Philippe   04/11/2010   \n",
       "107                          Yu Liu / Franz Bernd Lang   05/08/2010   \n",
       "112                     Yaoqing Shen / Gertraud Burger   01/04/2010   \n",
       "113                  Mathieu Lajoie / Nadia El-Mabrouk   05/05/2010   \n",
       "117                 Lilianne Dupuis / Normand Mousseau   05/04/2012   \n",
       "128         Béatrice Roure / Hervé Philippe/Franz Lang   06/10/2011   \n",
       "151               Mathieu Courcelles / Pierre Thibault   03/05/2012   \n",
       "159              Konstantin Bokov / Serguei Chteinberg   02/08/2013   \n",
       "161      Marie Pier Scott-Boyer / Christian Deschepper   03/09/2013   \n",
       "162                     Julie Hussin / Philip Awadalla   03/09/2013   \n",
       "165                          Malika Aïd / Sylvie Mader   03/03/2014   \n",
       "177                   Tetsu Ishii / Serguei Chteinberg   22/09/2014   \n",
       "178             Geneviève Galarneau / Guillaume Lettre   22/09/2014   \n",
       "185  Louis-Philippe Lemieux-Perreault / Marie-Pierr...   01/05/2014   \n",
       "186         Olivier Tremblay-Savard / Nadia El-Mabrouk   01/05/2014   \n",
       "202               Sandrine Moreira / / Gertraud Burger   2017-03-30   \n",
       "212                     Yohann Nédélec / Luis Barreiro   2018-03-12   \n",
       "216                 Diala Abd Rabbo / Stephen Michnick   2017-10-12   \n",
       "227                       Jee Eun Kang / Mohamed Hijri   2019-03-07   \n",
       "229                   Nicolas Girard / Pascale Legault   2019-03-07   \n",
       "230                Emmanuel Noutahi / Nadia El-Mabrouk   2019-03-07   \n",
       "234                    Tariq Daouda / Claude Perreault   2019-03-07   \n",
       "246       Louis-Philip Benoit-Bouvrette / Éric Lécuyer      à venir   \n",
       "\n",
       "            Programme                                   Titre_soutenance  \n",
       "61   Bio-informatique  Molecular protein function prediction using se...  \n",
       "62   Bio-informatique  Phylogenetic structural modeling of molecular ...  \n",
       "65   Bio-informatique  Étude de la structure tertiaire des motifs ARN...  \n",
       "86   Bio-informatique              Modeling heterotachy in phylogenetics  \n",
       "104  Bio-informatique    Statistical potentials for evolutionary studies  \n",
       "107  Bio-informatique  A phylogenomics approach to resolving fungal é...  \n",
       "112  Bio-informatique       In silico analysis of mitochondrial proteins  \n",
       "113  Bio-informatique  Approches algorithmiques pour l’inférence d’hi...  \n",
       "117  Bio-informatique  Développement d’une méthode multi-échelle pour...  \n",
       "128  Bio-informatique  Amélioration de l’exactitude de l’inférence ph...  \n",
       "151  Bio-informatique  Identification de nouveaux substrats des kinas...  \n",
       "159  Bio-informatique  Structural aspects of the ribosome evolution a...  \n",
       "161  Bio-informatique  Études de réseaux d’expression génique: utilit...  \n",
       "162  Bio-informatique  Genomic variation in recombination patterns: I...  \n",
       "165  Bio-informatique  Une nouvelle approche computationnelle pour la...  \n",
       "177  Bio-informatique  The  study of RNA tertiary interactions on tRN...  \n",
       "178  Bio-informatique  Genetic Determinants of Clinical Heterogeneity...  \n",
       "185  Bio-informatique  Approches bio-informatiques appliquées aux tec...  \n",
       "186  Bio-informatique  Étude de l’évolution des génomes par duplicati...  \n",
       "202  Bio-informatique       Décodage de l’expression de gènes cryptiques  \n",
       "212  Bio-informatique  Impact des variants génétiques sur la réponse ...  \n",
       "216  Bio-informatique  Beyond hairballs: Depicting complexity of a ki...  \n",
       "227  Bio-informatique  Novel Bioinformatics Programs for Taxonomical ...  \n",
       "229  Bio-informatique  Étude de la relation entre structure, dynamiqu...  \n",
       "230  Bio-informatique  Méthodes et algorithmes pour l’amélioration de...  \n",
       "234  Bio-informatique  L’usage des codons régule la présentation des ...  \n",
       "246  Bio-informatique  Caractérisation systématique des motifs de rég...  "
      ],
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Date</th>\n      <th>Etudiant_Directeur</th>\n      <th>Grade_obtenu</th>\n      <th>Programme</th>\n      <th>Titre_soutenance</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>61</th>\n      <td>17/12/2007</td>\n      <td>Sivakumar Kannan / Gertraud Burger</td>\n      <td>07/02/2008</td>\n      <td>Bio-informatique</td>\n      <td>Molecular protein function prediction using se...</td>\n    </tr>\n    <tr>\n      <th>62</th>\n      <td>14/12/2007</td>\n      <td>Nicolas Rodrigue / Hervé Philippe</td>\n      <td>07/02/2008</td>\n      <td>Bio-informatique</td>\n      <td>Phylogenetic structural modeling of molecular ...</td>\n    </tr>\n    <tr>\n      <th>65</th>\n      <td>07/08/2007</td>\n      <td>Emmanuelle Permal / François Major</td>\n      <td>04/10/2007</td>\n      <td>Bio-informatique</td>\n      <td>Étude de la structure tertiaire des motifs ARN...</td>\n    </tr>\n    <tr>\n      <th>86</th>\n      <td>27/08/2009</td>\n      <td>Yan Zhou / Hervé Philippe</td>\n      <td>08/10/2009</td>\n      <td>Bio-informatique</td>\n      <td>Modeling heterotachy in phylogenetics</td>\n    </tr>\n    <tr>\n      <th>104</th>\n      <td>28/09/2010</td>\n      <td>Claudia Laura Kleinman / Hervé Philippe</td>\n      <td>04/11/2010</td>\n      <td>Bio-informatique</td>\n      <td>Statistical potentials for evolutionary studies</td>\n    </tr>\n    <tr>\n      <th>107</th>\n      <td>23/06/2010</td>\n      <td>Yu Liu / Franz Bernd Lang</td>\n      <td>05/08/2010</td>\n      <td>Bio-informatique</td>\n      <td>A phylogenomics approach to resolving fungal é...</td>\n    </tr>\n    <tr>\n      <th>112</th>\n      <td>20/01/2010</td>\n      <td>Yaoqing Shen / Gertraud Burger</td>\n      <td>01/04/2010</td>\n      <td>Bio-informatique</td>\n      <td>In silico analysis of mitochondrial proteins</td>\n    </tr>\n    <tr>\n      <th>113</th>\n      <td>29/01/2010</td>\n      <td>Mathieu Lajoie / Nadia El-Mabrouk</td>\n      <td>05/05/2010</td>\n      <td>Bio-informatique</td>\n      <td>Approches algorithmiques pour l’inférence d’hi...</td>\n    </tr>\n    <tr>\n      <th>117</th>\n      <td>13/12/2011</td>\n      <td>Lilianne Dupuis / Normand Mousseau</td>\n      <td>05/04/2012</td>\n      <td>Bio-informatique</td>\n      <td>Développement d’une méthode multi-échelle pour...</td>\n    </tr>\n    <tr>\n      <th>128</th>\n      <td>25/07/2011</td>\n      <td>Béatrice Roure / Hervé Philippe/Franz Lang</td>\n      <td>06/10/2011</td>\n      <td>Bio-informatique</td>\n      <td>Amélioration de l’exactitude de l’inférence ph...</td>\n    </tr>\n    <tr>\n      <th>151</th>\n      <td>16/04/2012</td>\n      <td>Mathieu Courcelles / Pierre Thibault</td>\n      <td>03/05/2012</td>\n      <td>Bio-informatique</td>\n      <td>Identification de nouveaux substrats des kinas...</td>\n    </tr>\n    <tr>\n      <th>159</th>\n      <td>23/07/2013</td>\n      <td>Konstantin Bokov / Serguei Chteinberg</td>\n      <td>02/08/2013</td>\n      <td>Bio-informatique</td>\n      <td>Structural aspects of the ribosome evolution a...</td>\n    </tr>\n    <tr>\n      <th>161</th>\n      <td>26/06/2013</td>\n      <td>Marie Pier Scott-Boyer / Christian Deschepper</td>\n      <td>03/09/2013</td>\n      <td>Bio-informatique</td>\n      <td>Études de réseaux d’expression génique: utilit...</td>\n    </tr>\n    <tr>\n      <th>162</th>\n      <td>21/06/2013</td>\n      <td>Julie Hussin / Philip Awadalla</td>\n      <td>03/09/2013</td>\n      <td>Bio-informatique</td>\n      <td>Genomic variation in recombination patterns: I...</td>\n    </tr>\n    <tr>\n      <th>165</th>\n      <td>30/05/2013</td>\n      <td>Malika Aïd / Sylvie Mader</td>\n      <td>03/03/2014</td>\n      <td>Bio-informatique</td>\n      <td>Une nouvelle approche computationnelle pour la...</td>\n    </tr>\n    <tr>\n      <th>177</th>\n      <td>04/07/2014</td>\n      <td>Tetsu Ishii / Serguei Chteinberg</td>\n      <td>22/09/2014</td>\n      <td>Bio-informatique</td>\n      <td>The  study of RNA tertiary interactions on tRN...</td>\n    </tr>\n    <tr>\n      <th>178</th>\n      <td>25/06/2014</td>\n      <td>Geneviève Galarneau / Guillaume Lettre</td>\n      <td>22/09/2014</td>\n      <td>Bio-informatique</td>\n      <td>Genetic Determinants of Clinical Heterogeneity...</td>\n    </tr>\n    <tr>\n      <th>185</th>\n      <td>13/02/2014</td>\n      <td>Louis-Philippe Lemieux-Perreault / Marie-Pierr...</td>\n      <td>01/05/2014</td>\n      <td>Bio-informatique</td>\n      <td>Approches bio-informatiques appliquées aux tec...</td>\n    </tr>\n    <tr>\n      <th>186</th>\n      <td>22/01/2014</td>\n      <td>Olivier Tremblay-Savard / Nadia El-Mabrouk</td>\n      <td>01/05/2014</td>\n      <td>Bio-informatique</td>\n      <td>Étude de l’évolution des génomes par duplicati...</td>\n    </tr>\n    <tr>\n      <th>202</th>\n      <td>2016-11-29</td>\n      <td>Sandrine Moreira / / Gertraud Burger</td>\n      <td>2017-03-30</td>\n      <td>Bio-informatique</td>\n      <td>Décodage de l’expression de gènes cryptiques</td>\n    </tr>\n    <tr>\n      <th>212</th>\n      <td>2017-11-20</td>\n      <td>Yohann Nédélec / Luis Barreiro</td>\n      <td>2018-03-12</td>\n      <td>Bio-informatique</td>\n      <td>Impact des variants génétiques sur la réponse ...</td>\n    </tr>\n    <tr>\n      <th>216</th>\n      <td>2017-08-22</td>\n      <td>Diala Abd Rabbo / Stephen Michnick</td>\n      <td>2017-10-12</td>\n      <td>Bio-informatique</td>\n      <td>Beyond hairballs: Depicting complexity of a ki...</td>\n    </tr>\n    <tr>\n      <th>227</th>\n      <td>2018-12-07</td>\n      <td>Jee Eun Kang / Mohamed Hijri</td>\n      <td>2019-03-07</td>\n      <td>Bio-informatique</td>\n      <td>Novel Bioinformatics Programs for Taxonomical ...</td>\n    </tr>\n    <tr>\n      <th>229</th>\n      <td>2018-11-07</td>\n      <td>Nicolas Girard / Pascale Legault</td>\n      <td>2019-03-07</td>\n      <td>Bio-informatique</td>\n      <td>Étude de la relation entre structure, dynamiqu...</td>\n    </tr>\n    <tr>\n      <th>230</th>\n      <td>2018-11-06</td>\n      <td>Emmanuel Noutahi / Nadia El-Mabrouk</td>\n      <td>2019-03-07</td>\n      <td>Bio-informatique</td>\n      <td>Méthodes et algorithmes pour l’amélioration de...</td>\n    </tr>\n    <tr>\n      <th>234</th>\n      <td>2018-09-20</td>\n      <td>Tariq Daouda / Claude Perreault</td>\n      <td>2019-03-07</td>\n      <td>Bio-informatique</td>\n      <td>L’usage des codons régule la présentation des ...</td>\n    </tr>\n    <tr>\n      <th>246</th>\n      <td>Mardi 1er septembre 2020</td>\n      <td>Louis-Philip Benoit-Bouvrette / Éric Lécuyer</td>\n      <td>à venir</td>\n      <td>Bio-informatique</td>\n      <td>Caractérisation systématique des motifs de rég...</td>\n    </tr>\n  </tbody>\n</table>\n</div>"
     },
     "metadata": {},
     "execution_count": 7
    }
   ],
   "source": [
    "bioinfo_phd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "year=2021\n",
    "bioinfo_phd.to_csv(f'./{year}/AEBINUM_PHD_LIST_{year}.csv', sep=',', encoding='utf-8', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2. Create table of UdeM bioinformatics master students"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in range(len(master_links)):\n",
    "    if i == 0 :\n",
    "        all_master_student = get_table(master_links[i])\n",
    "    else:\n",
    "        tmp = get_table(master_links[i])\n",
    "        all_master_student = all_master_student.append(tmp, sort=True)\n",
    "## All student table\n",
    "all_master_student = all_master_student.dropna().reset_index().drop(columns=['index'])\n",
    "all_master_student['Etudiant_Directeur'] = all_master_student['Etudiant_Directeur'].apply(format_content)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "58\n"
     ]
    }
   ],
   "source": [
    "bioinfo_master = all_master_student[all_master_student['Programme'] == 'Bio-informatique']\n",
    "print(len(bioinfo_master))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "           Date                                 Etudiant_Directeur  \\\n",
       "50   07/12/2005               Geneviève Boucher / Normand Mousseau   \n",
       "77   21/12/2006  Mohamed Tikah Marrakchi / Sylvie Hamel / co: A...   \n",
       "82   07/11/2006  Valentin Guignon / Sylvie Hamel / co : Cedric ...   \n",
       "88   28/04/2006                        Tan Wang / Joelle Pelletier   \n",
       "93   17/12/2007      Audrey Noël / Pavel Hamet / co : Ettore Merlo   \n",
       "94   19/11/2007             Louis Philippe Lavoie / François Major   \n",
       "98   31/07/2007  Guy Larochelle / Hervé Philippe / co : Michel ...   \n",
       "99   03/07/2007                      Fabrice Baro / Hervé Philippe   \n",
       "100  27/05/2007              Sébastien Christin / Nadia El-Mabrouk   \n",
       "105  12/04/2007                Nicolas Saint-Onge / François Major   \n",
       "108  23/03/2007         Jean-François St-Pierre / Normand Mousseau   \n",
       "115  10/12/2008                   Wafae El Alaoui / Hervé Philippe   \n",
       "122  26/06/2008       Xiao Zhu / Traian Sulea / co : Robert Ménard   \n",
       "123  02/05/2008  Louis-Philippe Lemieux Perreault / Ettore Merl...   \n",
       "125  07/04/2008  Martin Smith / Gertraud Burger / co : Barbara ...   \n",
       "126  19/03/2008                    Pascal St-Onge / Daniel Sinnett   \n",
       "127  01/02/2008  François Gauthier / Pavel Hamet / co: Ettore M...   \n",
       "129  21/01/2008                   Félix Doyon / Serguei Chteinberg   \n",
       "133  02/11/2009                Diala Abd Rabbo / Christine Maugard   \n",
       "134  03/09/2009              Caroline Louis-Jeune / François Major   \n",
       "136  25/08/2009  Amandine Bemmo / Miklos Csuros / Co : Jacek Ma...   \n",
       "138  05/06/2009         Marie Pier Scott-Boyer / Sébastien Lemieux   \n",
       "142  20/04/2009                       Julie Hussin / Damian Labuda   \n",
       "150  22/11/2010                        Zia Mohaddes / Sylvie Hamel   \n",
       "159  02/11/2011                Emilio Damian Vello / Damian Labuda   \n",
       "160  03/10/2011  Asma Boufaden / Sylvie Mader / co : François M...   \n",
       "161  20/09/2011                    Éloi Mercier / Raphaël Gottardo   \n",
       "163  18/08/2011           Jean-Christophe Grenier / Hervé Philippe   \n",
       "164  14/07/2011              François Lefebvre / Sébastien Lemieux   \n",
       "168  15/05/2011              Alexandre Bramoullé / Pierre Thibault   \n",
       "170  11/04/2011                      Lucien Fabre / François Major   \n",
       "171  18/03/2011      Louis-Philip Benoit-Bouvrette / Miklos Csuros   \n",
       "187  10/09/2013                     Fatou Tao / Serguei Chteinberg   \n",
       "196  12/02/2013                 Raphaël Poujol / Nicolas Lartillot   \n",
       "204  07/10/2014  Jacques Boisvert / Paul Maddox / Sébastien Lem...   \n",
       "205  26/08/2014                        Alain Pacis / Luis Barreiro   \n",
       "216  12/02/2014                     Eric Fournier / Hervé Philippe   \n",
       "225  2015-10-06                 Séton Steven Bocco / Miklos Csuros   \n",
       "226  2015-09-03          Marie Armande Ang Houle / Philip Awadalla   \n",
       "230  2015-07-23              Jean-François Théroux / B. Franz Lang   \n",
       "234  2016-10-16                 Benjamin Matala / Stephen Michnick   \n",
       "245  2016-01-18          Sarah Cherkaoui / / Christine Des Rosiers   \n",
       "246  2016-01-08                  Amell El Korbi / / François Major   \n",
       "250  2017-12-13                  Assya Trofimov / Claude Perreault   \n",
       "258  2017-03-13                    Yann Ilboudo / Guillaume Lettre   \n",
       "259  2017-02-10                    Paméla Mehanna / Daniel Sinnett   \n",
       "263  2018-12-10                     Olivier Tastet / Luis Barreiro   \n",
       "266  2018-10-28                    Anh-Tien Ton / Adrian Serohijos   \n",
       "267  2018-10-19                   Philippe Malric / François Major   \n",
       "268  2018-08-30                Martineau Jean-Louis / Mark Samuels   \n",
       "270  2018-07-12               Caroline Labelle / Sébastien Lemieux   \n",
       "272  2018-04-16  Sandra Therrien-Laperrière / Christine Des Ros...   \n",
       "275  2018-02-22                      Matthew Sarrasin / Franz Lang   \n",
       "281  2019-11-12               Louis Gendron / / Sebastian Pechmann   \n",
       "282  2019-11-08                      Redha Attaoua / / Pavel Hamet   \n",
       "285  2020-12-08                    Cédric Roux / Daniel Sinnett /    \n",
       "288  2020-10-15     Sébastian Méric De Bellefon / Guillaume Lettre   \n",
       "293  2020-03-11   Pedro Do Couto Bordignon / Sebastian Pechmann /    \n",
       "\n",
       "    Grade_obtenu         Programme  \\\n",
       "50    12/01/2006  Bio-informatique   \n",
       "77    01/02/2007  Bio-informatique   \n",
       "82    07/12/2006  Bio-informatique   \n",
       "88    05/10/2006  Bio-informatique   \n",
       "93    07/02/2008  Bio-informatique   \n",
       "94    10/01/2008  Bio-informatique   \n",
       "98    06/09/2007  Bio-informatique   \n",
       "99    02/08/2007  Bio-informatique   \n",
       "100   02/08/2007  Bio-informatique   \n",
       "105   03/05/2007  Bio-informatique   \n",
       "108   03/05/2007  Bio-informatique   \n",
       "115   12/01/2009  Bio-informatique   \n",
       "122   03/07/2008  Bio-informatique   \n",
       "123   05/06/2008  Bio-informatique   \n",
       "125   01/05/2008  Bio-informatique   \n",
       "126   01/05/2008  Bio-informatique   \n",
       "127   13/03/2008  Bio-informatique   \n",
       "129   13/03/2008  Bio-informatique   \n",
       "133   03/12/2009  Bio-informatique   \n",
       "134   08/10/2009  Bio-informatique   \n",
       "136   05/11/2009  Bio-informatique   \n",
       "138   09/07/2009  Bio-informatique   \n",
       "142   04/06/2009  Bio-informatique   \n",
       "150   03/02/2011  Bio-informatique   \n",
       "159   05/01/2012  Bio-informatique   \n",
       "160   03/11/2011  Bio-informatique   \n",
       "161   01/12/2011  Bio-informatique   \n",
       "163   06/10/2011  Bio-informatique   \n",
       "164   03/11/2011  Bio-informatique   \n",
       "168   07/07/2011  Bio-informatique   \n",
       "170   05/05/2011  Bio-informatique   \n",
       "171   05/05/2011  Bio-informatique   \n",
       "187   10/12/2013  Bio-informatique   \n",
       "196   01/03/2013  Bio-informatique   \n",
       "204   19/02/2015  Bio-informatique   \n",
       "205   19/02/2015  Bio-informatique   \n",
       "216   20/05/2014  Bio-informatique   \n",
       "225   2016-03-17  Bio-informatique   \n",
       "226   2016-03-17  Bio-informatique   \n",
       "230   2015-10-15  Bio-informatique   \n",
       "234   2017-03-30  Bio-informatique   \n",
       "245   2016-03-17  Bio-informatique   \n",
       "246   2016-03-17  Bio-informatique   \n",
       "250   2018-03-12  Bio-informatique   \n",
       "258   2017-05-01  Bio-informatique   \n",
       "259   2017-05-01  Bio-informatique   \n",
       "263   2019-03-07  Bio-informatique   \n",
       "266   2019-03-07  Bio-informatique   \n",
       "267   2019-03-07  Bio-informatique   \n",
       "268   2019-03-07  Bio-informatique   \n",
       "270   2018-10-11  Bio-informatique   \n",
       "272   2018-06-27  Bio-informatique   \n",
       "275   2018-05-10  Bio-informatique   \n",
       "281   2020-03-18  Bio-informatique   \n",
       "282   2020-03-18  Bio-informatique   \n",
       "285      à venir  Bio-informatique   \n",
       "288      à venir  Bio-informatique   \n",
       "293      à venir  Bio-informatique   \n",
       "\n",
       "                                      Titre_soutenance  \n",
       "50   Exploration des mécanismes d’agrégation de pep...  \n",
       "77   Helix Explorer : une nouvelle base de données ...  \n",
       "82   Algorithme de comparaison de structures second...  \n",
       "88   Linker-scanning analysis of the HIV-1 integras...  \n",
       "93   Recherche de déterminants génomiques impliqués...  \n",
       "94                             Analyse de motifs d’ARN  \n",
       "98   Recherche d’une empreinte phylogénétique relié...  \n",
       "99   Phylogénie et transferts horizontaux de gènes ...  \n",
       "100  Recherche de snoRNA dans le génome de S. cerev...  \n",
       "105    MC-MAP, un nouvel outil d’intégration de motifs  \n",
       "108  Méthodes de simulations moléculaires accélérée...  \n",
       "115  Datation moléculaire et artefact dû à la satur...  \n",
       "122  Structural bioinformatics analysis of the fami...  \n",
       "123  Algorithme génétique spécifique à l’analyse de...  \n",
       "125  Insights into the function of short interspers...  \n",
       "126  Détection et caractérisation des interactions ...  \n",
       "127  Analyse de descendances ; une approche bio-inf...  \n",
       "129  Etude des interactions entre les boucles D et ...  \n",
       "133  Étude du transcriptome des cellules non tumora...  \n",
       "134  Représentation et recherche de motifs cyclique...  \n",
       "136  Performances de la puce exon et son applicatio...  \n",
       "138  Annotation des ARN non codants du génome du Ca...  \n",
       "142  Classes alléliques d’haplotypes et sélection p...  \n",
       "150  Modeling protein évolution using secondary str...  \n",
       "159  La cartographie des sites de régulation généti...  \n",
       "160  Prédiction de boucles de régulation associant ...  \n",
       "161  Développement d’outils pour l’analyse de donné...  \n",
       "163                         Phylogénomique des Archées  \n",
       "164  Comparaison des méthodes d’analyse de l’expres...  \n",
       "168  Identification des peptides du complexe majeur...  \n",
       "170  Étude structurale conformationnelle des toxine...  \n",
       "171  Analyse de la corrélation conditionnelle dériv...  \n",
       "187  Structural rules fot the formation of backbone...  \n",
       "196  Estimation des corrélations phylogénétiques en...  \n",
       "204  A robust algorithm for segmenting fluorescence...  \n",
       "205  Dynamic epigenetic changes in immune responses...  \n",
       "216  Évolution moléculaire : Un modèle Markov-modul...  \n",
       "225  Évolution à fine échelle des sites d’épissage ...  \n",
       "226  Typage de la classe génotypique du gène PRDM9 ...  \n",
       "230  Développement de méthodes d’assemblage de géno...  \n",
       "234  Une correction à l’échelle et progressive des ...  \n",
       "245  Développement d’une méthode bio-informatique p...  \n",
       "246  Identification de caractéristiques communes et...  \n",
       "250  Étude des signatures géniques dans un contexte...  \n",
       "258  The Genetics of Red Blood Cell Density, a Biom...  \n",
       "259  Caractérisation du microDNome et sa modulation...  \n",
       "263  Contrôle génétique de l’épissage alternatif da...  \n",
       "266  Multiscale Framework of Microbial Evolution to...  \n",
       "267  Prédiction et visualisation de la réactivité c...  \n",
       "268  Séquençage d’exomes chez des patients canadien...  \n",
       "270  Méthodologie pour l’analyse de données de crib...  \n",
       "272  Développement d’un outils bio-informatique pou...  \n",
       "275  An Approach to Improved Microbial Eukaryotic G...  \n",
       "281  Adaptation de la levure à la suite des perturb...  \n",
       "282  Investigation génétique de NAFLD dans le diabè...  \n",
       "285  Classification moléculaire des Tumeurs de Wilm...  \n",
       "288  Développement de méthodes bio-informatiques po...  \n",
       "293  An Analysis of Translation Heterogeneity in Ri...  "
      ],
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Date</th>\n      <th>Etudiant_Directeur</th>\n      <th>Grade_obtenu</th>\n      <th>Programme</th>\n      <th>Titre_soutenance</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>50</th>\n      <td>07/12/2005</td>\n      <td>Geneviève Boucher / Normand Mousseau</td>\n      <td>12/01/2006</td>\n      <td>Bio-informatique</td>\n      <td>Exploration des mécanismes d’agrégation de pep...</td>\n    </tr>\n    <tr>\n      <th>77</th>\n      <td>21/12/2006</td>\n      <td>Mohamed Tikah Marrakchi / Sylvie Hamel / co: A...</td>\n      <td>01/02/2007</td>\n      <td>Bio-informatique</td>\n      <td>Helix Explorer : une nouvelle base de données ...</td>\n    </tr>\n    <tr>\n      <th>82</th>\n      <td>07/11/2006</td>\n      <td>Valentin Guignon / Sylvie Hamel / co : Cedric ...</td>\n      <td>07/12/2006</td>\n      <td>Bio-informatique</td>\n      <td>Algorithme de comparaison de structures second...</td>\n    </tr>\n    <tr>\n      <th>88</th>\n      <td>28/04/2006</td>\n      <td>Tan Wang / Joelle Pelletier</td>\n      <td>05/10/2006</td>\n      <td>Bio-informatique</td>\n      <td>Linker-scanning analysis of the HIV-1 integras...</td>\n    </tr>\n    <tr>\n      <th>93</th>\n      <td>17/12/2007</td>\n      <td>Audrey Noël / Pavel Hamet / co : Ettore Merlo</td>\n      <td>07/02/2008</td>\n      <td>Bio-informatique</td>\n      <td>Recherche de déterminants génomiques impliqués...</td>\n    </tr>\n    <tr>\n      <th>94</th>\n      <td>19/11/2007</td>\n      <td>Louis Philippe Lavoie / François Major</td>\n      <td>10/01/2008</td>\n      <td>Bio-informatique</td>\n      <td>Analyse de motifs d’ARN</td>\n    </tr>\n    <tr>\n      <th>98</th>\n      <td>31/07/2007</td>\n      <td>Guy Larochelle / Hervé Philippe / co : Michel ...</td>\n      <td>06/09/2007</td>\n      <td>Bio-informatique</td>\n      <td>Recherche d’une empreinte phylogénétique relié...</td>\n    </tr>\n    <tr>\n      <th>99</th>\n      <td>03/07/2007</td>\n      <td>Fabrice Baro / Hervé Philippe</td>\n      <td>02/08/2007</td>\n      <td>Bio-informatique</td>\n      <td>Phylogénie et transferts horizontaux de gènes ...</td>\n    </tr>\n    <tr>\n      <th>100</th>\n      <td>27/05/2007</td>\n      <td>Sébastien Christin / Nadia El-Mabrouk</td>\n      <td>02/08/2007</td>\n      <td>Bio-informatique</td>\n      <td>Recherche de snoRNA dans le génome de S. cerev...</td>\n    </tr>\n    <tr>\n      <th>105</th>\n      <td>12/04/2007</td>\n      <td>Nicolas Saint-Onge / François Major</td>\n      <td>03/05/2007</td>\n      <td>Bio-informatique</td>\n      <td>MC-MAP, un nouvel outil d’intégration de motifs</td>\n    </tr>\n    <tr>\n      <th>108</th>\n      <td>23/03/2007</td>\n      <td>Jean-François St-Pierre / Normand Mousseau</td>\n      <td>03/05/2007</td>\n      <td>Bio-informatique</td>\n      <td>Méthodes de simulations moléculaires accélérée...</td>\n    </tr>\n    <tr>\n      <th>115</th>\n      <td>10/12/2008</td>\n      <td>Wafae El Alaoui / Hervé Philippe</td>\n      <td>12/01/2009</td>\n      <td>Bio-informatique</td>\n      <td>Datation moléculaire et artefact dû à la satur...</td>\n    </tr>\n    <tr>\n      <th>122</th>\n      <td>26/06/2008</td>\n      <td>Xiao Zhu / Traian Sulea / co : Robert Ménard</td>\n      <td>03/07/2008</td>\n      <td>Bio-informatique</td>\n      <td>Structural bioinformatics analysis of the fami...</td>\n    </tr>\n    <tr>\n      <th>123</th>\n      <td>02/05/2008</td>\n      <td>Louis-Philippe Lemieux Perreault / Ettore Merl...</td>\n      <td>05/06/2008</td>\n      <td>Bio-informatique</td>\n      <td>Algorithme génétique spécifique à l’analyse de...</td>\n    </tr>\n    <tr>\n      <th>125</th>\n      <td>07/04/2008</td>\n      <td>Martin Smith / Gertraud Burger / co : Barbara ...</td>\n      <td>01/05/2008</td>\n      <td>Bio-informatique</td>\n      <td>Insights into the function of short interspers...</td>\n    </tr>\n    <tr>\n      <th>126</th>\n      <td>19/03/2008</td>\n      <td>Pascal St-Onge / Daniel Sinnett</td>\n      <td>01/05/2008</td>\n      <td>Bio-informatique</td>\n      <td>Détection et caractérisation des interactions ...</td>\n    </tr>\n    <tr>\n      <th>127</th>\n      <td>01/02/2008</td>\n      <td>François Gauthier / Pavel Hamet / co: Ettore M...</td>\n      <td>13/03/2008</td>\n      <td>Bio-informatique</td>\n      <td>Analyse de descendances ; une approche bio-inf...</td>\n    </tr>\n    <tr>\n      <th>129</th>\n      <td>21/01/2008</td>\n      <td>Félix Doyon / Serguei Chteinberg</td>\n      <td>13/03/2008</td>\n      <td>Bio-informatique</td>\n      <td>Etude des interactions entre les boucles D et ...</td>\n    </tr>\n    <tr>\n      <th>133</th>\n      <td>02/11/2009</td>\n      <td>Diala Abd Rabbo / Christine Maugard</td>\n      <td>03/12/2009</td>\n      <td>Bio-informatique</td>\n      <td>Étude du transcriptome des cellules non tumora...</td>\n    </tr>\n    <tr>\n      <th>134</th>\n      <td>03/09/2009</td>\n      <td>Caroline Louis-Jeune / François Major</td>\n      <td>08/10/2009</td>\n      <td>Bio-informatique</td>\n      <td>Représentation et recherche de motifs cyclique...</td>\n    </tr>\n    <tr>\n      <th>136</th>\n      <td>25/08/2009</td>\n      <td>Amandine Bemmo / Miklos Csuros / Co : Jacek Ma...</td>\n      <td>05/11/2009</td>\n      <td>Bio-informatique</td>\n      <td>Performances de la puce exon et son applicatio...</td>\n    </tr>\n    <tr>\n      <th>138</th>\n      <td>05/06/2009</td>\n      <td>Marie Pier Scott-Boyer / Sébastien Lemieux</td>\n      <td>09/07/2009</td>\n      <td>Bio-informatique</td>\n      <td>Annotation des ARN non codants du génome du Ca...</td>\n    </tr>\n    <tr>\n      <th>142</th>\n      <td>20/04/2009</td>\n      <td>Julie Hussin / Damian Labuda</td>\n      <td>04/06/2009</td>\n      <td>Bio-informatique</td>\n      <td>Classes alléliques d’haplotypes et sélection p...</td>\n    </tr>\n    <tr>\n      <th>150</th>\n      <td>22/11/2010</td>\n      <td>Zia Mohaddes / Sylvie Hamel</td>\n      <td>03/02/2011</td>\n      <td>Bio-informatique</td>\n      <td>Modeling protein évolution using secondary str...</td>\n    </tr>\n    <tr>\n      <th>159</th>\n      <td>02/11/2011</td>\n      <td>Emilio Damian Vello / Damian Labuda</td>\n      <td>05/01/2012</td>\n      <td>Bio-informatique</td>\n      <td>La cartographie des sites de régulation généti...</td>\n    </tr>\n    <tr>\n      <th>160</th>\n      <td>03/10/2011</td>\n      <td>Asma Boufaden / Sylvie Mader / co : François M...</td>\n      <td>03/11/2011</td>\n      <td>Bio-informatique</td>\n      <td>Prédiction de boucles de régulation associant ...</td>\n    </tr>\n    <tr>\n      <th>161</th>\n      <td>20/09/2011</td>\n      <td>Éloi Mercier / Raphaël Gottardo</td>\n      <td>01/12/2011</td>\n      <td>Bio-informatique</td>\n      <td>Développement d’outils pour l’analyse de donné...</td>\n    </tr>\n    <tr>\n      <th>163</th>\n      <td>18/08/2011</td>\n      <td>Jean-Christophe Grenier / Hervé Philippe</td>\n      <td>06/10/2011</td>\n      <td>Bio-informatique</td>\n      <td>Phylogénomique des Archées</td>\n    </tr>\n    <tr>\n      <th>164</th>\n      <td>14/07/2011</td>\n      <td>François Lefebvre / Sébastien Lemieux</td>\n      <td>03/11/2011</td>\n      <td>Bio-informatique</td>\n      <td>Comparaison des méthodes d’analyse de l’expres...</td>\n    </tr>\n    <tr>\n      <th>168</th>\n      <td>15/05/2011</td>\n      <td>Alexandre Bramoullé / Pierre Thibault</td>\n      <td>07/07/2011</td>\n      <td>Bio-informatique</td>\n      <td>Identification des peptides du complexe majeur...</td>\n    </tr>\n    <tr>\n      <th>170</th>\n      <td>11/04/2011</td>\n      <td>Lucien Fabre / François Major</td>\n      <td>05/05/2011</td>\n      <td>Bio-informatique</td>\n      <td>Étude structurale conformationnelle des toxine...</td>\n    </tr>\n    <tr>\n      <th>171</th>\n      <td>18/03/2011</td>\n      <td>Louis-Philip Benoit-Bouvrette / Miklos Csuros</td>\n      <td>05/05/2011</td>\n      <td>Bio-informatique</td>\n      <td>Analyse de la corrélation conditionnelle dériv...</td>\n    </tr>\n    <tr>\n      <th>187</th>\n      <td>10/09/2013</td>\n      <td>Fatou Tao / Serguei Chteinberg</td>\n      <td>10/12/2013</td>\n      <td>Bio-informatique</td>\n      <td>Structural rules fot the formation of backbone...</td>\n    </tr>\n    <tr>\n      <th>196</th>\n      <td>12/02/2013</td>\n      <td>Raphaël Poujol / Nicolas Lartillot</td>\n      <td>01/03/2013</td>\n      <td>Bio-informatique</td>\n      <td>Estimation des corrélations phylogénétiques en...</td>\n    </tr>\n    <tr>\n      <th>204</th>\n      <td>07/10/2014</td>\n      <td>Jacques Boisvert / Paul Maddox / Sébastien Lem...</td>\n      <td>19/02/2015</td>\n      <td>Bio-informatique</td>\n      <td>A robust algorithm for segmenting fluorescence...</td>\n    </tr>\n    <tr>\n      <th>205</th>\n      <td>26/08/2014</td>\n      <td>Alain Pacis / Luis Barreiro</td>\n      <td>19/02/2015</td>\n      <td>Bio-informatique</td>\n      <td>Dynamic epigenetic changes in immune responses...</td>\n    </tr>\n    <tr>\n      <th>216</th>\n      <td>12/02/2014</td>\n      <td>Eric Fournier / Hervé Philippe</td>\n      <td>20/05/2014</td>\n      <td>Bio-informatique</td>\n      <td>Évolution moléculaire : Un modèle Markov-modul...</td>\n    </tr>\n    <tr>\n      <th>225</th>\n      <td>2015-10-06</td>\n      <td>Séton Steven Bocco / Miklos Csuros</td>\n      <td>2016-03-17</td>\n      <td>Bio-informatique</td>\n      <td>Évolution à fine échelle des sites d’épissage ...</td>\n    </tr>\n    <tr>\n      <th>226</th>\n      <td>2015-09-03</td>\n      <td>Marie Armande Ang Houle / Philip Awadalla</td>\n      <td>2016-03-17</td>\n      <td>Bio-informatique</td>\n      <td>Typage de la classe génotypique du gène PRDM9 ...</td>\n    </tr>\n    <tr>\n      <th>230</th>\n      <td>2015-07-23</td>\n      <td>Jean-François Théroux / B. Franz Lang</td>\n      <td>2015-10-15</td>\n      <td>Bio-informatique</td>\n      <td>Développement de méthodes d’assemblage de géno...</td>\n    </tr>\n    <tr>\n      <th>234</th>\n      <td>2016-10-16</td>\n      <td>Benjamin Matala / Stephen Michnick</td>\n      <td>2017-03-30</td>\n      <td>Bio-informatique</td>\n      <td>Une correction à l’échelle et progressive des ...</td>\n    </tr>\n    <tr>\n      <th>245</th>\n      <td>2016-01-18</td>\n      <td>Sarah Cherkaoui / / Christine Des Rosiers</td>\n      <td>2016-03-17</td>\n      <td>Bio-informatique</td>\n      <td>Développement d’une méthode bio-informatique p...</td>\n    </tr>\n    <tr>\n      <th>246</th>\n      <td>2016-01-08</td>\n      <td>Amell El Korbi / / François Major</td>\n      <td>2016-03-17</td>\n      <td>Bio-informatique</td>\n      <td>Identification de caractéristiques communes et...</td>\n    </tr>\n    <tr>\n      <th>250</th>\n      <td>2017-12-13</td>\n      <td>Assya Trofimov / Claude Perreault</td>\n      <td>2018-03-12</td>\n      <td>Bio-informatique</td>\n      <td>Étude des signatures géniques dans un contexte...</td>\n    </tr>\n    <tr>\n      <th>258</th>\n      <td>2017-03-13</td>\n      <td>Yann Ilboudo / Guillaume Lettre</td>\n      <td>2017-05-01</td>\n      <td>Bio-informatique</td>\n      <td>The Genetics of Red Blood Cell Density, a Biom...</td>\n    </tr>\n    <tr>\n      <th>259</th>\n      <td>2017-02-10</td>\n      <td>Paméla Mehanna / Daniel Sinnett</td>\n      <td>2017-05-01</td>\n      <td>Bio-informatique</td>\n      <td>Caractérisation du microDNome et sa modulation...</td>\n    </tr>\n    <tr>\n      <th>263</th>\n      <td>2018-12-10</td>\n      <td>Olivier Tastet / Luis Barreiro</td>\n      <td>2019-03-07</td>\n      <td>Bio-informatique</td>\n      <td>Contrôle génétique de l’épissage alternatif da...</td>\n    </tr>\n    <tr>\n      <th>266</th>\n      <td>2018-10-28</td>\n      <td>Anh-Tien Ton / Adrian Serohijos</td>\n      <td>2019-03-07</td>\n      <td>Bio-informatique</td>\n      <td>Multiscale Framework of Microbial Evolution to...</td>\n    </tr>\n    <tr>\n      <th>267</th>\n      <td>2018-10-19</td>\n      <td>Philippe Malric / François Major</td>\n      <td>2019-03-07</td>\n      <td>Bio-informatique</td>\n      <td>Prédiction et visualisation de la réactivité c...</td>\n    </tr>\n    <tr>\n      <th>268</th>\n      <td>2018-08-30</td>\n      <td>Martineau Jean-Louis / Mark Samuels</td>\n      <td>2019-03-07</td>\n      <td>Bio-informatique</td>\n      <td>Séquençage d’exomes chez des patients canadien...</td>\n    </tr>\n    <tr>\n      <th>270</th>\n      <td>2018-07-12</td>\n      <td>Caroline Labelle / Sébastien Lemieux</td>\n      <td>2018-10-11</td>\n      <td>Bio-informatique</td>\n      <td>Méthodologie pour l’analyse de données de crib...</td>\n    </tr>\n    <tr>\n      <th>272</th>\n      <td>2018-04-16</td>\n      <td>Sandra Therrien-Laperrière / Christine Des Ros...</td>\n      <td>2018-06-27</td>\n      <td>Bio-informatique</td>\n      <td>Développement d’un outils bio-informatique pou...</td>\n    </tr>\n    <tr>\n      <th>275</th>\n      <td>2018-02-22</td>\n      <td>Matthew Sarrasin / Franz Lang</td>\n      <td>2018-05-10</td>\n      <td>Bio-informatique</td>\n      <td>An Approach to Improved Microbial Eukaryotic G...</td>\n    </tr>\n    <tr>\n      <th>281</th>\n      <td>2019-11-12</td>\n      <td>Louis Gendron / / Sebastian Pechmann</td>\n      <td>2020-03-18</td>\n      <td>Bio-informatique</td>\n      <td>Adaptation de la levure à la suite des perturb...</td>\n    </tr>\n    <tr>\n      <th>282</th>\n      <td>2019-11-08</td>\n      <td>Redha Attaoua / / Pavel Hamet</td>\n      <td>2020-03-18</td>\n      <td>Bio-informatique</td>\n      <td>Investigation génétique de NAFLD dans le diabè...</td>\n    </tr>\n    <tr>\n      <th>285</th>\n      <td>2020-12-08</td>\n      <td>Cédric Roux / Daniel Sinnett /</td>\n      <td>à venir</td>\n      <td>Bio-informatique</td>\n      <td>Classification moléculaire des Tumeurs de Wilm...</td>\n    </tr>\n    <tr>\n      <th>288</th>\n      <td>2020-10-15</td>\n      <td>Sébastian Méric De Bellefon / Guillaume Lettre</td>\n      <td>à venir</td>\n      <td>Bio-informatique</td>\n      <td>Développement de méthodes bio-informatiques po...</td>\n    </tr>\n    <tr>\n      <th>293</th>\n      <td>2020-03-11</td>\n      <td>Pedro Do Couto Bordignon / Sebastian Pechmann /</td>\n      <td>à venir</td>\n      <td>Bio-informatique</td>\n      <td>An Analysis of Translation Heterogeneity in Ri...</td>\n    </tr>\n  </tbody>\n</table>\n</div>"
     },
     "metadata": {},
     "execution_count": 11
    }
   ],
   "source": [
    "bioinfo_master"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "year=2021\n",
    "bioinfo_master.to_csv(f'./{year}/AEBINUM_MASTER_LIST_{year}.csv', sep=',', encoding='utf-8', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5-final"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}